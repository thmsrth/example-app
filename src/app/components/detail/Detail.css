.Detail__container {
  overflow-y: auto;
  overflow-x: hidden;
  max-height: calc(100vh - 75px);
}

.Detail__dashboard {
  display: flex;
  flex-wrap: wrap;
}

.Detail__actions {
  margin: 0;
}

.Detail__icon {
  margin-right: var(--space);
}

.DetailBack {
  position: absolute;
  top: 0;
  left: 0;
}
