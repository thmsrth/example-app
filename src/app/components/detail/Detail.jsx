import React from 'react';

import dataItemType from '../types/dataItem';

import DataStateNotifier from '../../../global/components/DataStateNotifier';

import DetailHeader from './DetailHeader';
import DetailInfos from './DetailInfos';

import BeersBox from './beers/BeersBox';

import './Detail.css';

const Detail = ({
  item, dataUpdating, ...otherProps
}) => {
  if (item) {
    return (
      <DataStateNotifier
        {...otherProps}
      >
        <div className="Detail">
          <DetailHeader
            {...otherProps}
            item={item}
          />
          <div className="Detail__container">
            <div className="Detail__dashboard">
              <DetailInfos
                {...otherProps}
                item={item}
              />
              <BeersBox
                {...otherProps}
                item={item}
              />
            </div>
          </div>
        </div>
      </DataStateNotifier>
    );
  }
  return null;
};

export default Detail;

Detail.propTypes = {
  item: dataItemType,
};

Detail.defaultProps = {
  item: {},
};
