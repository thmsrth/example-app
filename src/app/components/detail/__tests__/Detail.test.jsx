import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import staticData from '../../../static/data.json';
import Detail from '../Detail';

global.permissions = {};

const item = staticData._embedded.foods[0];
const props = { item };

describe('Detail', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<Detail {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
