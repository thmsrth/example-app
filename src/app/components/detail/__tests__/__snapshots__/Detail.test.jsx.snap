// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail should render the component correctly 1`] = `
<DataStateNotifier
  dataAdding=""
  dataDeleting=""
  dataError=""
  dataLoading=""
  dataUpdating=""
  resetDataError={[Function]}
>
  <div
    className="Detail"
  >
    <DetailHeader
      dataFilters={Object {}}
      dataPaginationOffset={0}
      dataPaginationSize={25}
      dataSortParam="id,ASC"
      item={
        Object {
          "calories": 266,
          "createdAt": "2018-10-28T14:48:54.01293",
          "createdBy": "thomas",
          "fat": 34,
          "id": 1,
          "name": "Pizza",
          "protein": 0.6,
          "status": "UNHEALTHY",
          "updatedAt": null,
          "updatedBy": null,
        }
      }
    />
    <div
      className="Detail__container"
    >
      <div
        className="Detail__dashboard"
      >
        <DetailInfos
          item={
            Object {
              "calories": 266,
              "createdAt": "2018-10-28T14:48:54.01293",
              "createdBy": "thomas",
              "fat": 34,
              "id": 1,
              "name": "Pizza",
              "protein": 0.6,
              "status": "UNHEALTHY",
              "updatedAt": null,
              "updatedBy": null,
            }
          }
        />
        <BeersBox
          beersError=""
          beersLoading=""
          item={
            Object {
              "calories": 266,
              "createdAt": "2018-10-28T14:48:54.01293",
              "createdBy": "thomas",
              "fat": 34,
              "id": 1,
              "name": "Pizza",
              "protein": 0.6,
              "status": "UNHEALTHY",
              "updatedAt": null,
              "updatedBy": null,
            }
          }
        />
      </div>
    </div>
  </div>
</DataStateNotifier>
`;
